users/views.py

```diff
@@ -0,0 +1,188 @@
+                response["message"] = "Successfully created user."
+                response["code"] = status.HTTP_201_CREATED
+            else:
+                response["message"] = "Email already exists."
+                if not valid_name:
+                    response["message"] = "Name can not be blank."
+                response["code"] = status.HTTP_400_BAD_REQUEST
+        except KeyError:
+            response["message"] = "Required fields don't exist."
+            response["code"] = status.HTTP_400_BAD_REQUEST
+        except IntegrityError as integrity_error:
+            response["message"] = str(integrity_error)
+            response["code"] = status.HTTP_409_CONFLICT
+        except ValidationError as validation_error:
+            response["message"] = " ".join(validation_error)
+            response["code"] = status.HTTP_400_BAD_REQUEST
+
+        return Response(response)
+
+    @authentication_classes(TokenAuthentication)
+    @permission_classes(IsAuthenticated)
+    def patch(self, request):
+        """
+        This function updates data of existing user
+        Args:
+            request(HttpRequest): Value containing request data
+        Returns:
+            (Response): A json object containing message and code
+        """
+        try:
+            user = request.user
+            if "email" in request.POST:
+                email = request.POST["email"]
+                validate_email(email)
+                if not User.objects.filter(email=email).exists():
+                    user.email = request.POST["email"]
+            if "name" in request.POST:
+                name = request.POST["name"]
+                if name not in ["", " "]:
+                    user.name = name
+                else:
+                    return Response(
+                        {
+                            "message": "Name is not valid",
+                            "code": status.HTTP_400_BAD_REQUEST,
+                        }
+                    )
+            if "password" in request.POST:
+                validate_password(request.POST["password"])
+                user.password = make_password(request.POST["password"])
+            user.save()
+            return Response(
+                {"message": "Changes updated successfully", "code": status.HTTP_200_OK}
+            )
+        except ValidationError as error:
+            return Response(
+                {"message": " ".join(error), "code": status.HTTP_400_BAD_REQUEST}
+            )
+
+    @authentication_classes(TokenAuthentication)
+    @permission_classes(IsAuthenticated)
+    def delete(self, request):
+        """
+        This function deletes a user from database
+        Args:
+            request(HttpRequest): Value containing request data
+        Returns:
+            (Response): A json object containing message and code
+        """
+        user = request.user
+        response = {}
+        if user.is_superuser:
```

Comment:
Can we create a `SuperUserPermission` class and use this check there, and do something like `@permission_classes(IsAuthenticated, IsSuperUser)` annotation. We will be able to reuse it at multiple places otherwise we will have to apply this check whenever there is something we want to do with super user permission.

Reply:
Can you elaborate a little more on this

Reply:
To simplify, Can we create custom permission for `is_SuperUser` instead of checking `if user.is_superuser:` in here?